s5a/__init__.py

```diff
@@ -6,80 +6,72 @@
-    def __init__(self, longitude, latitude, value, timestamp, quality):
-        self.longitude = longitude
-        self.latitude = latitude
-        self.value = value
-        self.timestamp = timestamp
-        self.quality = quality
+    :param ncfile: path of the file to be read
+    :type ncfile: string
+    :return: a geopandas geodataframe containing
+        the groundpixel information as points
+    :rtype: geopandas.GeoDataFrame
+    """
 
-    def __repr__(self):
-        return (
-            f'longitude={self.longitude} latitude={self.latitude} '
-            f'value={self.value} timestamp={self.timestamp} '
-            f'quality={self.quality}')
+    # read in data
+    with netCDF4.Dataset(ncfile, 'r') as f:
+        variables = f.groups['PRODUCT'].variables
+        data = variables['carbonmonoxide_total_column'][:][0]
+        longitude = variables['longitude'][:][0]
+        latitude = variables['latitude'][:][0]
+        quality = variables['qa_value'][:][0]
+        deltatime = variables['delta_time'][:][0]
+        meta_data = f.__dict__
 
+    # get some metadata
+    # use mask from MaskedArray to filter values
+    mask = numpy.logical_not(data.mask)
+    n_lines = data.shape[0]  # number of scan lines
+    pixel_per_line = data.shape[1]  # number of pixels per line
+    time_reference = meta_data['time_reference_seconds_since_1970']
 
-class Scan():
-    """Object to hold arrays from an nc file.
-    """
-    def __init__(self, filepath):
-        self.filepath = filepath
-        self.raw_data = RawData(filepath)
-        self.time_reference = datetime.utcfromtimestamp(
-            self.raw_data.meta_data['time_reference_seconds_since_1970'])
-        self.points = []
-        shape = self.raw_data.data.shape
-        for i in range(shape[0]):
-            for j in range(shape[1]):
-                timestamp = self.time_reference + timedelta(
-                        milliseconds=int(self.raw_data.deltatime[i]))
-                self.points.append(Point(
-                    longitude=float(self.raw_data.longitude[i, j]),
-                    latitude=float(self.raw_data.latitude[i, j]),
-                    value=float(self.raw_data.data[i, j]),
-                    quality=float(self.raw_data.quality[i, j]),
-                    timestamp=timestamp,
-                ))
+    # convert deltatime to timestamps
+    deltatime_arr = numpy.repeat(
+        deltatime, pixel_per_line).reshape(n_lines, -1)
+    deltatime_arr = deltatime_arr[mask]  # filter for missing data
+    # add (milli-)seconds since 1970
+    deltatime_arr = numpy.add(deltatime_arr, time_reference * 1000)
+    timestamps = pandas.to_datetime(deltatime_arr, utc=True, unit='ms')
 
-    def filter_by_quality(self, minimal_quality):
-        """Filter points of the Scan by quality.
+    # convert data to geodataframe
+    return geopandas.GeoDataFrame({
+        'timestamp': timestamps,
+        'quality': quality[mask],
+        'data': data[mask]
```

Comment:
```suggestion
        'value': data[mask]
```
Maybe for consistency with the database entries, we should name this `value` as well

Reply:
I agree. [Done](https://github.com/emissions-api/sentinel5algorithms/pull/17/commits/3f01b51eb0bb0bad1bb9fa70a49c9f6b66b8feb8).